internal/biz/resources/resources.go

```diff
@@ -118,12 +122,19 @@ func (uc *Usecase) Upsert(ctx context.Context, m *model.Resource) (*model.Resour
 		}
 
 		if uc.ListenManager != nil {
-			timeoutCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
-			defer cancel()
 
-			err = subscription.BlockForNotification(timeoutCtx)
-			if err != nil {
-				return nil, err
+			// read after write functionality is enabled globally.
+			// And executed if request specifies or
+			// request came from service provider in allowlist
+			if uc.ReadAfterWriteEnabled && (requestReadAfterWrite || isSPInAllowlist(m, uc.ReadAfterWriteAllowlist)) {
```

Comment:
We will need to also add this condition to the other block that nil checks the ListenManager above. We don't want to subscribe if it's not requested by the client.

Perhaps we could throw it into a bool for share-ability.

```
readAfterWriteEnabled := uc.ListenManager != nil && uc.ReadAfterWriteEnabled && (requestReadAfterWrite || isSPInAllowlist(m, uc.ReadAfterWriteAllowlist))
```

Reply:
Missed the subscription part thanks! That bool's a great idea